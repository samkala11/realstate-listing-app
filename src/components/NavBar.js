import React from 'react';


const NavBar = () => {

    return (
        <div className="nav-bar">
        
            <h2> Nav </h2>

        </div>
    ) 
}

export default NavBar;