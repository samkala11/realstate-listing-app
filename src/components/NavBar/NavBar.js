import React, {useEffect} from 'react';
import Paper from '@material-ui/core/Paper';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';  
import { useHistory } from 'react-router-dom';

const NavBar = () => {

    const [value, setValue] = React.useState(0);
    const history = useHistory();

    const handleChange = (event, newValue) => {
      setValue(newValue);

        if (newValue === 0) {
            console.log('going to home ');
            history.push('/')
        } else if (newValue === 1) {
            console.log('going to listingss ');
            history.push('/listings')
        } else if (newValue === 2) {
            console.log('going to saved listingss ');
            history.push('/saved-listings')
        }   
    };

    useEffect(() => {
        let  currentHref = window.location.href
        if (currentHref.substr(currentHref.length - 1) === '/') {
            return;
        } else if (currentHref.substr(currentHref.length - 9) === '/listings') {
            setValue(1)
        } else if (currentHref.substr(currentHref.length - 15) === '/saved-listings') {
            setValue(2)
        }
    }, [])

    return (
        <div data-testid='nav-bar' className="nav-bar">
            <Paper id="nav-bar-container" square>
                <Tabs
                    value={value}
                    indicatorColor="primary"
                    textColor="primary"
                    onChange={handleChange}
                    aria-label="disabled tabs example"
                >
                    <Tab data-testid='home-tab' label="Home" />
                    <Tab data-testid='listings-tab' label="Property Listings" />
                    <Tab data-testid='saved-listings-tab' label="Saved Listings" />
                    
                </Tabs>
            </Paper>

        </div>
    ) 
}

export default NavBar;