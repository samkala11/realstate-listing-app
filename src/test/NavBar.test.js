import { render, screen, cleanup } from '@testing-library/react';
import NavBar from '../components/NavBar/NavBar';
import renderer from 'react-test-renderer';

afterEach(cleanup);

test('Renders NavBar page successfully', () => {

    render(<NavBar />);
    const navBarTitle = screen.getByTestId('nav-bar');
    expect(navBarTitle).toBeInTheDocument();
  
});

test('NavBar page show all 3 tabs successfully', () => {

    render(<NavBar />);
    const homeTab = screen.getByTestId('home-tab');
    expect(homeTab).toBeInTheDocument();
    expect(homeTab).toHaveTextContent('Home')

    const listingsTab = screen.getByTestId('listings-tab');
    expect(listingsTab).toBeInTheDocument();
    expect(listingsTab).toHaveTextContent('Property Listings')

    const savedListingsTab = screen.getByTestId('saved-listings-tab');
    expect(savedListingsTab).toBeInTheDocument();
    expect(savedListingsTab).toHaveTextContent('Saved Listings')
  
});

test('NavBar page matches snapshot', () => {
    const tree = renderer.create(<NavBar />).toJSON();
    expect(tree).toMatchSnapshot();
})