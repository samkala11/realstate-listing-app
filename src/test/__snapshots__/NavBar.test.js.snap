// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar page matches snapshot 1`] = `
<div
  className="nav-bar"
  data-testid="nav-bar"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1"
    id="nav-bar-container"
  >
    <div
      className="MuiTabs-root"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": null,
            "overflow": "hidden",
          }
        }
      >
        <div
          aria-label="disabled tabs example"
          className="MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
            data-testid="home-tab"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Home
            </span>
            <span
              className="PrivateTabIndicator-root-9 PrivateTabIndicator-colorPrimary-10 MuiTabs-indicator"
              style={Object {}}
            />
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
            data-testid="listings-tab"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Property Listings
            </span>
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
            data-testid="saved-listings-tab"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Saved Listings
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
